lib/fluent/config/configure_proxy.rb

```diff
@@ -63,21 +63,36 @@ def multi?
       end
 
       def final?
-        @final
+        @final == true
       end
 
       def merge(other) # self is base class, other is subclass
         return merge_for_finalized(other) if self.final?
 
-        options = {
-          param_name: other.param_name,
-          required: (other.required.nil? ? self.required : other.required),
-          multi: (other.multi.nil? ? self.multi : other.multi)
-        }
+        if overwrite?(other, :required)
+          raise ConfigError, "subclass cannot overwrite base class's definition"
```

Comment:
And, could you add header to show that it's a bug of plugin, not users' misconfigurations (e.g., "BUG:")